python/ray/tune/progress_reporter.py

```diff
@@ -38,43 +29,126 @@ def report(self, trial_runner):
+                    trial_runner.get_trials(),
+                    metric_columns=self.metric_columns,
+                    fmt=self.fmt,
+                    max_rows=self.max_progress_rows))
+        if self.max_error_rows > 0:
+            messages.append(
+                trial_errors_str(
+                    trial_runner.get_trials(),
+                    fmt=self.fmt,
+                    max_rows=self.max_error_rows))
+        return messages
+
+
+class JupyterNotebookReporter(TuneReporterBase):
+    """Jupyter notebook-friendly Reporter that can update display in-place."""
+
+    def __init__(self,
+                 overwrite,
+                 metric_columns=None,
+                 max_progress_rows=20,
+                 max_error_rows=20):
+        """Initializes a new JupyterNotebookReporter.
+
+        Args:
+            overwrite (bool): Flag for overwriting the last reported progress.
+            metric_columns (Dict[str,str]|List[str]): Names of metrics to
+                include in progress table. If this is a dict, the keys should
+                be metric names and the values should be the displayed names.
+                If this is a list, the metric name is used directly.
+            max_progress_rows (int): Maximum number of progress rows to print
+                in the progress table. Defaults to 20.
+            max_error_rows (int): Maximum number of error rows to print in the
+                error table. Defaults to 20.
+        """
+        super(JupyterNotebookReporter,
+              self).__init__(metric_columns, max_progress_rows, max_error_rows)
+        self.overwrite = overwrite
+
+    def report(self, trial_runner):
+        delim = "<br>"
         from IPython.display import clear_output
         from IPython.core.display import display, HTML
         if self.overwrite:
             clear_output(wait=True)
-        display(HTML(delim.join(messages) + delim))
+        display(HTML(delim.join(self._messages(trial_runner)) + delim))
 
 
-class CLIReporter(ProgressReporter):
+class CLIReporter(TuneReporterBase):
+    """Command-line reporter"""
+
+    def __init__(self,
+                 metric_columns=None,
+                 max_progress_rows=20,
+                 max_error_rows=20):
+        """Initializes a CLIReporter.
+
+        Args:
+            metric_columns (Dict[str,str]|List[str]): Names of metrics to
+                include in progress table. If this is a dict, the keys should
+                be metric names and the values should be the displayed names.
+                If this is a list, the metric name is used directly.
+            max_progress_rows (int): Maximum number of progress rows to print
+                in the progress table. Defaults to 20.
+            max_error_rows (int): Maximum number of error rows to print in the
+                error table. Defaults to 20.
+        """
+        super(CLIReporter, self).__init__(metric_columns, max_progress_rows,
```

Comment:
```suggestion
        return super(CLIReporter, self).__init__(metric_columns, max_progress_rows,
```

Reply:
is this the convention even for __init__ even though it doesnt return anything?

Reply:
Good question. I don't actually know what the generally agreed convention is. But here if your super changes s.t. it returns something, you'd have to add it anyway. And since you're only using the init for default parameters, I think it might be better to just return whatever the super returns.

Reply:
hmm so i checked and  it looks like `__init__` is required to return None so in this particular scenario i'd just keep it as is